sharktank_models/test_suite/ireers_tools/artifacts.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import Any, Callable, Collection, Dict, Union
+import functools
+from pathlib import Path
+from tqdm import tqdm
+import urllib.parse
+import urllib.request
+import os
+from azure.storage.blob import BlobClient, BlobProperties
+import hashlib
+import mmap
+import re
+import logging
+
+logger = logging.getLogger(__name__)
+# Adjust logging levels.
+logging.basicConfig(level=logging.INFO)
+for log_name, log_obj in logging.Logger.manager.loggerDict.items():
+    if log_name.startswith("azure"):
+        logging.getLogger(log_name).setLevel(logging.WARNING)
+
+
+def show_progress(t):
+    last_b = [0]
+
+    def update_to(b=1, bsize=1, tsize=None):
+        if tsize is not None:
+            t.total = tsize
+        t.update((b - last_b[0]) * bsize)
+        last_b[0] = b
+
+    return update_to
+
+
+@functools.cache
+def get_artifact_root_dir() -> Path:
+    root_path = os.getenv("IREE_TEST_FILES", default=str(Path.cwd())) + "/artifacts"
```

Comment:
Similar comment about `Path.cwd()`. Could root on `sharktank_models/test_suite/` (get path to the current script file then walk up the tree as needed)